### PR TITLE
Update packaging and clean up Makefile

### PR DESCRIPTION
**Pull request by freeekanayaka**
This branch changes mainly the makefiles in order to:

1) Have the install target also install the test TLS certficate that
   fake-juju uses by default

2) Install not only the fake-jujud binary but also the fake-juju
   one (which the CLI).

3) Streamline makefile.mk targets a bit more, not perfect but an
   improvement (also, some legacy variables have been moved to the
   1.25 and 2.0.0 trees, which will eventually be dropped).

There is also a small change in the fake-juju CLI to make the
bootstrap command automatically switch to the newly created
controller (like it does in reality). This shaves a tiny bit of
boilerplate from Python integration tests too.

### CODE REVIEW

File: common/makefile.mk [hunk 1]
```diff
@@ -55,21 +44,41 @@ test: $(JUJU_VERSION)
...
+# Copy fake-juju Go code extensions to the upstream source tree
+.PHONY: overlay
+overlay: src
+	# First clean up the upstream source tree from any overlay symlink
+	# we might have created in previous runs.
+	find src/ -name *-fakejuju.go | xargs rm -f
 
+	# Then create new symlinks against our overlay Go source files.
+	for name in $(shell find ../common/core/ -name "*.go"); \
+		do ln -s $$(pwd)/$$name src/$$(echo $$name | cut -d / -f 4-); \
```
Comment (freeekanayaka) [inline common/makefile.mk]: Essentially it's stripping the first 3 segments of a non-absolute path:

```
~$ echo ../common/core/github.com/juju/testing/mgo-fakejuju.go | cut -d / -f 4-
github.com/juju/testing/mgo-fakejuju.go
```

Perhaps I could replace it with a sed if that's more clear:

```
~$ echo ../common/core/github.com/juju/testing/mgo-fakejuju.go | sed -e "s|^../common/core/||"
github.com/juju/testing/mgo-fakejuju.go
```

File: common/makefile.mk [hunk 2]
```diff
@@ -55,21 +44,41 @@ test: $(JUJU_VERSION)
...
+
+# Apply quilt patches to the upstream source tree
+.PHONY: patch
+patch: src
+	quilt applied > /dev/null 2>&1 || quilt push -a
+
+# Extract the upstream tarball
+.PHONY: src
+src: $(JUJU_TARBALL)
+	test -e src || tar --strip=1 -z -xf $(JUJU_TARBALL)
```
Comment (freeekanayaka) [inline common/makefile.mk]: Unfortunately, that doesn't work, as figured out on IRC.

File: common/service/run.go [hunk 3]
```diff
@@ -46,13 +46,18 @@ func RunFakeJuju() int {
 	mongo := flags.Int("mongo", 0, "Optional external MongoDB port to use (default is to spawn a new instance on a random free port)")
 	port := flags.Int("port", 17099, "The port the API server will listent to")
 	series := flags.String("series", "xenial", "Ubuntu series")
+	debug := flags.Bool("debug", false, "Enable debug logging")
 	flags.Parse(os.Args[1:])
 
+	level := loggo.INFO
+	if *debug {
+		level = loggo.DEBUG
```
Comment (freeekanayaka) [inline common/service/run.go]: I don't think there is a way to do that in the standard library:

https://golang.org/pkg/flag/

File: common/makefile.mk [hunk 4]
```diff
@@ -55,21 +44,41 @@ test: $(JUJU_VERSION)
...
-clean:
-	rm -f $(JUJU_TARBALL)
-	rm -rf $(JUJU_SRC)
-	rm -f $(JUJU_VERSION)
+# Copy fake-juju Go code extensions to the upstream source tree
+.PHONY: overlay
+overlay: src
+	# First clean up the upstream source tree from any overlay symlink
+	# we might have created in previous runs.
+	find src/ -name *-fakejuju.go | xargs rm -f
```
Comment (freeekanayaka) [inline common/makefile.mk]: Fixed.

File: debian/control [hunk 5]
```diff
@@ -16,13 +16,15 @@ Build-Depends: debhelper (>= 9),
                python-testtools,
                python-twisted-core (>= 16.0.0),
                python-txjuju,
+               python-txfixtures,
```
Comment (freeekanayaka) [inline debian/control]: txFixed.

File: debian/control [hunk 6]
```diff
@@ -16,13 +16,15 @@ Build-Depends: debhelper (>= 9),
...
                python-yaml,
                wget
 
 Package: fake-juju
 Architecture: any
 Depends: ${shlibs:Depends}, ${misc:Depends},
          juju-mongodb,
+         juju-mongodb3.2,
```
Comment (freeekanayaka) [inline debian/control]: Juju uses mongo 3.2 by default. Using it in fake-juju makes it closer to the real thing. I plan to drop juju-mongodb once we drop the old fake-juju code.

File: debian/control [hunk 7]
```diff
@@ -36,5 +38,6 @@ Depends: ${python:Depends}, ${misc:Depends}, ${extra:Depends}, ${shlibs:Depends}
          python-testtools,
          python-twisted-core (>= 16.0.0),
          python-txjuju,
+         python-txfixtures,
```
Comment (freeekanayaka) [inline debian/control]: txFixed

File: common/makefile.mk (hunk 2 above)
Comment (sparkiegeek) [inline common/makefile.mk]: :+1:

File: common/makefile.mk (hunk 1 above)
Comment (sparkiegeek) [inline common/makefile.mk]: :+1: I think the `sed` makes it clearer

File: common/service/run.go (hunk 3 above)
Comment (sparkiegeek) [inline common/service/run.go]: :disappointed: fair enough

File: debian/control (hunk 6 above)
Comment (sparkiegeek) [inline debian/control]: OK, as long as we long term settle on one of them I'm happy. Thanks, :bowing_man:

**Pull request merged**